src/main/java/gregtech/common/covers/filter/OreDictionaryItemFilter.java

```diff
@@ -1,77 +1,80 @@
+import gregtech.api.mui.GTGuis;
 import gregtech.api.unification.OreDictUnifier;
 import gregtech.api.unification.stack.ItemVariantMap;
 import gregtech.api.unification.stack.MultiItemVariantMap;
 import gregtech.api.unification.stack.SingleItemVariantMap;
-import gregtech.api.util.function.BooleanConsumer;
 import gregtech.api.util.oreglob.OreGlob;
 import gregtech.api.util.oreglob.OreGlobCompileResult;
 import gregtech.common.covers.filter.oreglob.impl.ImpossibleOreGlob;
 import gregtech.common.gui.widget.HighlightedTextField;
-import gregtech.common.gui.widget.orefilter.ItemOreFilterTestSlot;
-import gregtech.common.gui.widget.orefilter.OreGlobCompileStatusWidget;
+import gregtech.common.gui.widget.orefilter.OreFilterTestSlot;
 
+import net.minecraft.client.resources.I18n;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
-import net.minecraft.network.PacketBuffer;
 import net.minecraft.util.text.TextFormatting;
 
+import com.cleanroommc.modularui.api.drawable.IKey;
+import com.cleanroommc.modularui.drawable.UITexture;
+import com.cleanroommc.modularui.screen.ModularPanel;
+import com.cleanroommc.modularui.screen.Tooltip;
+import com.cleanroommc.modularui.utils.Color;
+import com.cleanroommc.modularui.value.sync.BooleanSyncValue;
+import com.cleanroommc.modularui.value.sync.GuiSyncManager;
+import com.cleanroommc.modularui.value.sync.StringSyncValue;
+import com.cleanroommc.modularui.widget.Widget;
+import com.cleanroommc.modularui.widgets.SlotGroupWidget;
+import com.cleanroommc.modularui.widgets.ToggleButton;
+import com.cleanroommc.modularui.widgets.layout.Column;
+import com.cleanroommc.modularui.widgets.layout.Row;
 import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import java.util.function.BooleanSupplier;
 import java.util.function.Consumer;
 
 public class OreDictionaryItemFilter extends ItemFilter {
 
     private final Map<Item, ItemVariantMap.Mutable<Boolean>> matchCache = new Object2ObjectOpenHashMap<>();
     private final SingleItemVariantMap<Boolean> noOreDictMatch = new SingleItemVariantMap<>();
 
-    protected String expression = "";
-
     private OreGlob glob = ImpossibleOreGlob.getInstance();
-    private boolean error;
+    private OreGlobCompileResult result;
+    private final OreDictionaryItemFilterReader filterReader;
 
-    private boolean caseSensitive;
-    /**
-     * {@code false} requires any of the entry to be match in order for the match to be success, {@code true} requires
-     * all entries to match
-     */
-    private boolean matchAll;
+    public OreDictionaryItemFilter(ItemStack stack) {
+        this.filterReader = new OreDictionaryItemFilterReader(stack, 0);
+        setFilterReader(this.filterReader);
+        recompile();
```

Comment:
Do we need to call this here?

Reply:
recompile needs to be called for both server and client, and filters are instantiated on both sides. the call in `readFromNBT()` could be removed now that i think about it

Reply:
it's been removed now